python/remage/geometry_benchmark/cli.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import annotations
+
+import argparse
+import ast
+import tempfile
+from pathlib import Path
+
+from ..cli import remage_run
+from .summary_generator import SummaryGenerator
+
+
+def generate_output_file_path(geometry_path: Path, output_dir: Path) -> Path:
+    """Generate the output file path based on the geometry file name.
+
+    Parameters
+    ----------
+    geometry_path : Path
+        Path to the geometry file.
+    output_dir : Path
+        Directory where the output file will be stored.
+
+    Returns
+    -------
+    Path
+        Full path to the output file with .lh5 extension.
+    """
+    filename = geometry_path.absolute()
+    output_file = output_dir / filename.name.replace(".gdml", ".lh5")
+    return output_file
+
+def generate_macro(args) -> str:
+    """Generate a macro file for the remage geometry benchmark.
+
+    This function creates a macro file based on the provided command-line
+    arguments, which will be used to configure the remage simulation for the
+    geometry benchmark.
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        Parsed command-line arguments.
+
+    Returns
+    -------
+    str
+        Path to the generated macro file.
+    """
+
+    geometry_path = Path(args.geometry)
+    filename = geometry_path.absolute()
+    output_dir = Path(args.output_dir)
+    output_file = generate_output_file_path(geometry_path, output_dir)
+    if issubclass(type(args.grid_increments), str) and args.grid_increments:
+        grid_increments = ast.literal_eval(args.grid_increments)
+        increment_x = grid_increments.get('x', args.grid_increment)
+        increment_y = grid_increments.get('y', args.grid_increment)
+        increment_z = grid_increments.get('z', args.grid_increment)
+    else:
+        increment_x = args.grid_increment
+        increment_y = args.grid_increment
+        increment_z = args.grid_increment
+    n_events = args.num_events
+
+
+
+    macro_content = f"""
+/RMG/Geometry/IncludeGDMLFile {filename}
+
+/RMG/Output/ActivateOutputScheme Benchmark
+/RMG/Output/FileName {output_file}
+/run/initialize
+
+/RMG/Generator/Select Benchmark
+
+/RMG/Generator/Benchmark/IncrementX {increment_x} mm
+/RMG/Generator/Benchmark/IncrementY {increment_y} mm
+/RMG/Generator/Benchmark/IncrementZ {increment_z} mm
+
+/run/beamOn {n_events}
+    """
+    
+    if args.dry_run:
+        print("Dry run enabled. Generated macro content:")
+        print(macro_content)
+        return ""
+
+    macro_file_path = tempfile.NamedTemporaryFile(mode="w", delete=False, suffix=".mac")
+    macro_file_path.write(macro_content)
+    macro_file_path.close()
+    return str(macro_file_path.name)
+
+
+def remage_geometry_benchmark_cli() -> int:
```

Comment:
add an optional args parameter, so that people can script this/test this in the test suite